Add a workaround for invalid aborts until the next tar release 
which has this fixed properly.  Thanks to Jim Faulkner for 
reporting and tracking this down !

http://lists.gnu.org/archive/html/bug-tar/2005-04/msg00012.html
http://bugs.gentoo.org/show_bug.cgi?id=87540

Benno Schulenberg says to update tests too:
http://bugs.gentoo.org/show_bug.cgi?id=90742

Index: src/buffer.c
===================================================================
RCS file: /cvsroot/tar/tar/src/buffer.c,v
retrieving revision 1.81
retrieving revision 1.82
diff -u -r1.81 -r1.82
--- src/buffer.c	4 Apr 2005 18:21:55 -0000	1.81
+++ src/buffer.c	7 Apr 2005 17:26:11 -0000	1.82
@@ -653,8 +653,8 @@
 	record_start++;
 
       if (strlen (real_s_name) > NAME_FIELD_SIZE)
-	FATAL_ERROR ((0, 0,
-		      _("%s: file name too long to be stored in a GNU multivolume header"),
+	WARN ((0, 0,
+	      _("%s: file name too long to be stored in a GNU multivolume header, truncated"),
 		      quotearg_colon (real_s_name)));
 
       memset (record_start, 0, BLOCKSIZE);
--- tests/multiv03.at
+++ tests/multiv03.at
@@ -33,8 +33,7 @@
 BFILE=`awk 'BEGIN { for (i = 0; i < 101; i++) printf "b"; exit; }'`
 
 cat > ../experr <<EOF
-tar: $BFILE: file name too long to be stored in a GNU multivolume header
-tar: Error is not recoverable: exiting now
+tar: $BFILE: file name too long to be stored in a GNU multivolume header, truncated
 EOF
 
 cat > ../expout <<EOF
@@ -52,7 +51,7 @@
 genfile --length 15360 > $BFILE
 tar -M -L 10 -c -f arch.1 -f arch.2 $BFILE 
 ],
-[2],
+[0],
 [expout],
 [experr],
 [],[],[gnu, oldgnu])
